content/en/docs/code/cri-api-dev-policies.md

```diff
@@ -0,0 +1,126 @@
+---
+content_type: "reference"
+title: Kubernetes feature development and container runtimes
+weight: 55
+---
+
+CRI is a plugin interface which enables kubelet to use a wide variety of container runtimes,
+without the need to recompile. CRI consists of a protocol buffers and gRPC API.
+Read more about CRI API at [kubernetes docs](https://kubernetes.io/docs/concepts/architecture/cri/).
+
+The mechanics of a feature development that requires new CRI APIs is covered in
+documentation on CRI API [feature-development](/docs/code/cri-api-version-skew-policy#feature-development).
+This article declares policies for developing new Kubernetes features
+that require CRI API changes. The goal of these policies is to ensure great user
+experience for people trying the new feature early, adopting it when it is
+enabled by default, and relying on it as a GA functionality.
+
+{{% alert title="Notice" color="warning" %}}
+This document establishes development policies for a new features requiring
+adherence to the policy by two currently participating runtime. This is not
+a commitment from Kubernetes community to support those these features. Also
+the list of participating container runtimes may change depending on the community
+engagement.
+{{% /alert %}}
+
+## Participating container runtimes
+
+Features and CRI API are supposed to be portable and generic and not limited to
+a specific container runtime. However at this moment SIG Node requires every
+feature to work on two container runtimes: [containerd](https://containerd.io/)
+and [CRI-O](https://cri-o.io/). These are two runtimes that are tested as part
+of a Kubernetes development and release process to refer features portability.
+When this document refers to two container runtimes, it assumes both -
+containerd and CRI-O. If any other container runtimes begin working actively
+with the Kubernetes community, or one of existing container runtimes will stop
+engagement, this document will be updated.
+
+## Same maturity level (for beta and GA)
+
+Implementation of an API needed for a Kubernetes feature in a container runtime
+**must** be at least the same maturity as in k8s at a moment of Kubernetes release.
+This is similar to the [deprecation policy](https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecating-a-feature-or-behavior)
+when one feature is replacing another.
+
+With containerd and CRI-O it means that for the GA features, there should be a
+release of containerd and CRI-O implementing APIs needed for the feature in
+those container runtimes. And those APIs **must not** be marked as experimental
+features ([containerd experimental features](https://containerd.io/releases/#experimental-features)).
+For the beta, neither container runtime has a notion of beta feature or release and
+realistically the same maturity criteria applies as for GA.
```

Comment:
Beta graduations may come with additional CRI changes. I assume container runtimes should adapt to them and document accordingly which feature graduation state they support.

Reply:
yes. Do you feel this needs to be spelled out explicitly?